src/master-playlist-controller.js

```diff
@@ -816,6 +818,43 @@ export class MasterPlaylistController extends videojs.EventTarget {
   }
 
   /**
+   * fill our internal list of Captions Tracks with data from
+   * the master playlist or use a default
+   *
+   * @private
+   */
+  fillClosedCaptionTracks_() {
+    let master = this.master();
+    let mediaGroups = master.mediaGroups || {};
+
+    for (let mediaGroup in mediaGroups['CLOSED-CAPTIONS']) {
+      if (!this.closedCaptionGroups_.groups[mediaGroup]) {
+        this.closedCaptionGroups_.groups[mediaGroup] = [];
+      }
+
+      for (let label in mediaGroups['CLOSED-CAPTIONS'][mediaGroup]) {
+        let properties = mediaGroups['CLOSED-CAPTIONS'][mediaGroup][label];
+
+        this.closedCaptionGroups_.groups[mediaGroup].push(
+          videojs.mergeOptions({ id: label }, properties));
+
+        if (typeof this.closedCaptionGroups_.tracks[label] === 'undefined') {
+          let track = this.tech_.addRemoteTextTrack({
+            id: properties.instreamId,
+            kind: 'captions',
+            enabled: false,
+            language: properties.language,
+            label
+          }, true).track;
```

Comment:
If we change this parameter from `true` to `false`, video.js will take care of removing the track on source changes